docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+  - [Testcase : Deployment usecases and fault handling scenario's](#testcase--deployment-usecases-and-fault-handling-scenarios)
+    - [Link flap on an interface with macsec configured](#link-flap-on-an-interface-with-macsec-configured)
+    - [Link flap on a portchannel member which has macsec configured](#link-flap-on-a-portchannel-member-which-has-macsec-configured)
+    - [MACsec session cannot be established under wrong MKA configuration](#macsec-session-cannot-be-established-under-wrong-mka-configuration)
+    - [Config reload done on DUT with macsec configuration](#config-reload-done-on-dut-with-macsec-configuration)
+    - [Everflow, port mirroring on macsec enabled interfaces](#everflow-port-mirroring-on-macsec-enabled-interfaces)
+  - [Testcase : Macsec scenario's for multi-asic, multi-dut](#testcase--macsec-scenarios-for-multi-asic-multi-dut)
+  - [Testcase : Scale tests](#testcase--scale-tests)
+    - [Large number of interfaces having macsec enabled on the DUT/linecard](#large-number-of-interfaces-having-macsec-enabled-on-the-dutlinecard)
+    - [Simultaneous rekeying in all macsec sessions](#simultaneous-rekeying-in-all-macsec-sessions)
+    - [Macsec enabled on all interfaces and the DUT is rebooted](#macsec-enabled-on-all-interfaces-and-the-dut-is-rebooted)
+
+## Testbed
+
+```txt
++-------------------------------------------------------------------------+
+|                                                                         |
+| DUT                                                                     |
+|                                                                         |
++-------+-------------------+-----------------+--------------+----------+-+
+        *                   *                 |              |          |
+        *                   *                 |              |          |
+        *.........          *.........        +......        +......    |
+        *        :          *        :        |     :        |     :    |
+        *        :          *        :        |     :        |     :    |
+  +-----+------+ :    +-----+------+ :    +---+---+ :    +---+---+ :    |
+  |VM0         | :    |VM1         | :    |VM2    | :    |VM3    | :    |
+  |(Controlled)| :    |(Controlled)| :    |       | :    |       | :    |
+  +------------+ :    +------------+ :    +-------+ :    +-------+ :    |
+                 :                   :              :              :    |
++----------------+-------------------+--------------+--------------+----+-+
+|                                                                         |
+| PTF                                                                     |
+|                                                                         |
++-------------------------------------------------------------------------+
+
+
+-----       normal link
+.....       injected link
+*****       protected link
+VM<->DUT    up link
+PTF<->DUT   down link
```

Comment:
This is specific to T0 topology. Incase of T2 the downlink can go to another VM. Is there a plan to cover test that part? where unprotected traffic goes from T1 node to T2 and T2 sends encrypted traffic to T3 ?

Reply:
I have a test case that send a unprotected traffic from PTF, and DUT will forward the packets to VM0(MACsec enabled VM). Does this case cover that you mentioned?
I believe in the most time, the T2 is a larger scale topo than the current testbed topology(T0). We are truly working on building that larger topology that will be a hybrid mode(vSONiC mixed with cEOS) for lesser resource consumption.